File: yyevac/src/main/cpp/bean/optionparams.cpp
//
// Created by zengjiale on 2022/9/22.
//

#include "optionparams.h"

OptionParams::~OptionParams() {
    mp4Address.clear();
}

OptionParams *OptionParams::parse(const char *json) {
    if (json == nullptr) return this;

    JSON_Value *root_value = json_parse_string(json);
    JSON_Value_Type jsonValueType = json_value_get_type(root_value);
    if (jsonValueType == JSONObject) {
        JSON_Object *jsonObject = json_value_get_object(root_value);
        if (json_object_has_value(jsonObject, "playerId")) {
            playerId = int(json_object_get_number(jsonObject, "playerId"));
        }
        if (json_object_has_value(jsonObject, "frameRate")) {
            fps = int(json_object_get_number(jsonObject, "frameRate"));
        } else {
            fps = 30;
        }
        if (json_object_has_value(jsonObject, "playCount")) {
            playCount = int(json_object_get_number(jsonObject, "playCount"));
        } else {
            playCount = 1;
        }
        if (json_object_has_value(jsonObject, "isEnableH264")) {
            isEnableH264 = int(json_object_get_boolean(jsonObject, "isEnableH264"));
        } else {
            isEnableH264 = true;
        }

        if (json_object_has_value(jsonObject, "isEnableH265")) {
            isEnableH265 = int(json_object_get_boolean(jsonObject, "isEnableH265"));
        } else {
            isEnableH265 = false;
        }
        if (json_object_has_value(jsonObject, "isMute")) {
            isMute = int(json_object_get_boolean(jsonObject, "isMute"));
        } else {
            isMute = true;
        }
        if (json_object_has_value(jsonObject, "usingSurfaceView")) {
            usingSurfaceView = int(json_object_get_boolean(jsonObject, "usingSurfaceView"));
        }
        if (json_object_has_value(jsonObject, "mp4Address")) {
            mp4Address.clear();
            mp4Address = json_object_get_string(jsonObject, "mp4Address");
        } else {
            mp4Address = "";
        }
        if (json_object_has_value(jsonObject, "scaleType")) {
            scaleType = int(json_object_get_number(jsonObject, "scaleType"));
        } else {
            scaleType = 1;
        }
        if (json_object_has_value(jsonObject, "disableAudio")) {
            disableAudio = bool(json_object_get_boolean(jsonObject, "disableAudio"));
        } else {
            disableAudio = true;
        }
    }
    return this;
}

File: yyevac/src/main/java/com/yy/yyevav2/params/OptionParams.kt
package com.yy.yyevav2.params

import org.json.JSONObject

class OptionParams {
    var frameRate = 30
    var playCount = 1
    var isEnableH265 = false
    var isEnableH264 = false
    var isMute = false
    var usingSurfaceView = false
    var isRemoteService = true  //使用多进程
    var isMainProcess = true // 是否主进程
    var useFbo = true
    var mp4Address = ""
    var scaleType = 1   // 1=>裁剪居中， 2=>全屏拉伸  3=》原资源大小
    var filterType = "" //高清算法 hermite lagrange
        set(value) {
            if (value.isNotEmpty()) {
                useFbo = true //一定要使用fbo
            }
            field = value
        }

    fun toJson(): String {
        val json = JSONObject()
        json.put("frameRate", frameRate)
        json.put("playCount", playCount)
        json.put("isEnableH265", isEnableH265)
        json.put("isEnableH264", isEnableH264)
        json.put("isMute", isMute)
        json.put("usingSurfaceView", usingSurfaceView)
        json.put("isRemoteService", isRemoteService)
        json.put("isMainProcess", isMainProcess)
        json.put("useFbo", useFbo)
        json.put("mp4Address", mp4Address)
        json.put("scaleType", scaleType)
        json.put("filterType", filterType)
        return json.toString()
    }
}